tests/xharness/Xharness.Tests/Logging/Tests/LogFileTest.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.IO;
+using Moq;
+using NUnit.Framework;
+using Xharness.Logging;
+
+namespace Xharness.Tests.Logging.Tests {
+
+	[TestFixture]
+	public class LogFileTest {
+
+		string path;
+		string description;
+
+		Mock<ILogs> logs;
+
+		[SetUp]
+		public void SetUp ()
+		{
+			description = "My log";
+			path = Path.GetTempFileName ();
+			logs = new Mock<ILogs> ();
+			File.Delete (path); // delete the empty file
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			if (File.Exists (path))
+				File.Delete (path);
+			path = null;
+			description = null;
+			logs = null;
+		}
+
+		[Test]
+		public void ConstructorTest ()
+		{
+			using (var log = new LogFile (logs.Object, description, path)) {
+				Assert.AreEqual (description, log.Description, "description");
+				Assert.AreEqual (path, log.FullPath, "path");
+				Assert.AreSame (logs.Object, log.Logs, "logs");
+			}
+		}
+
+		[Test]
+		public void ConstructorNullILogsTest ()
+		{
+			Assert.DoesNotThrow (() => {
+				using (var log = new LogFile (null, description, path)) ;
+			});
+		}
+
+		[Test]
+		public void ConstructorNullPathTest ()
+		{
+			Assert.Throws<ArgumentNullException> (() => { var log = new LogFile (logs.Object, description, null); });
+		}
+
+		[Test]
+		public void ConstructorNullDescriptionTest ()
+		{
+			Assert.DoesNotThrow (() => {
+				using (var log = new LogFile (logs.Object, null, path)) ;
+			});
+		}
```

Comment:
Not sure these tests bring any actual value (apart from code coverage points)

Reply:
Is mainly code coverage. Make sure that if we start making it needed, that test fail and we know we change something that will break the code.